test/ws/authorizationWebSocketHandlerSpec.ts

```diff
@@ -0,0 +1,260 @@
+import { WebSocketServer } from 'ws';
+import WebSocket from 'ws';
+import { AuthorizationWebSocketHandler } from '../../src/ws/authorization-websocket-handler';
+import { BookmarkManager } from '../../src/ws/bookmark-manager';
+import { InverseSpecificationEngine } from '../../src/ws/inverse-specification-engine';
+import { FactEnvelope, FactRecord, FactReference } from '../../src/storage';
+import { Specification } from '../../src/specification/specification';
+import { serializeGraph } from '../../src/http/serializer';
+
+// Mock invertSpecification to return both add and remove operations
+jest.mock('../../src/specification/inverse', () => {
+  const dummySpec: Specification = { given: [{ name: 'g', type: 'T' }], matches: [], projection: { type: 'composite', components: [] } };
+  return {
+    invertSpecification: jest.fn(() => ([
+      { inverseSpecification: dummySpec, operation: 'add', givenSubset: [], parentSubset: [], path: '', resultSubset: [] },
+      { inverseSpecification: dummySpec, operation: 'remove', givenSubset: [], parentSubset: [], path: '', resultSubset: [] },
+    ]))
+  };
+});
+
+jest.setTimeout(15000);
+
+describe('AuthorizationWebSocketHandler', () => {
+  let wss: WebSocketServer;
+
+  beforeAll(async () => {
+    wss = new WebSocketServer({ port: 0 });
+    await new Promise<void>(resolve => wss.once('listening', () => resolve()));
+  });
+
+  afterAll(async () => {
+    await new Promise<void>(resolve => wss.close(() => resolve()));
+  });
+
+  test('sends BOOK sync on SUB when server has newer bookmark', async () => {
+    const address = wss.address();
```

Comment:
I'm concerned about the quality of these end-to-end tests given that they rely heavily upon mocks and stubs. Consider what would be necessary to integrate more of the Jinaga logic into these tests. For example, can the client connect using the full JinagaClient class configured with a wsEndpoint setting, rather than creating a WebSockets connection object directly?